src/AppBundle/v2/Assembler/ClientAssembler.php

```diff
@@ -2,52 +2,92 @@
 
 namespace AppBundle\v2\Assembler;
 
+use AppBundle\v2\Assembler\Report\ReportAssemblerInterface;
 use AppBundle\v2\DTO\ClientDto;
 
 class ClientAssembler
 {
+    /** @var ReportAssemblerInterface  */
+    private $reportDtoAssembler;
+
+    /** @var NdrAssembler */
+    private $ndrDtoAssembler;
+
+    /**
+     * @param ReportAssemblerInterface $reportDtoAssembler
+     * @param NdrAssembler $ndrDtoAssembler
+     */
+    public function __construct(ReportAssemblerInterface $reportDtoAssembler, NdrAssembler $ndrDtoAssembler)
+    {
+        $this->reportDtoAssembler = $reportDtoAssembler;
+        $this->ndrDtoAssembler = $ndrDtoAssembler;
+    }
+
     /**
      * @param array $data
      * @return ClientDto
      */
     public function assembleFromArray(array $data)
     {
-        $this->throwExceptionIfMissingRequiredData($data);
-
-        return new ClientDto(
-            $data['id'],
-            $data['case_number'],
-            $data['firstname'],
-            $data['lastname'],
-            $data['email'],
-            $data['report_count'],
-            $data['ndr_id']
-        );
+        $dto = new ClientDto();
```

Comment:
Just out of curiousity, why remove these params from the constructor and have the object constructed outside the class?